scalding-core/src/main/scala/com/twitter/scalding/estimation/memory/MemoryEstimatorStepStrategy.scala

```diff
@@ -0,0 +1,109 @@
+package com.twitter.scalding.estimation.memory
+
+import cascading.flow.{ Flow, FlowStep, FlowStepStrategy }
+import com.twitter.algebird.Monoid
+import com.twitter.scalding.estimation.{ Estimator, FallbackEstimatorMonoid, FlowStrategyInfo, MemoryEstimatorConfig }
+import com.twitter.scalding.{ Config, StringUtility }
+import java.util.{ List => JList }
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.mapred.JobConf
+import org.slf4j.LoggerFactory
+import scala.collection.JavaConverters._
+
+object MemoryEstimatorStepStrategy extends FlowStepStrategy[JobConf] {
+
+  private val LOG = LoggerFactory.getLogger(this.getClass)
+
+  implicit val estimatorMonoid = new FallbackEstimatorMonoid[MemoryEstimate]
+
+  /**
+   * Make memory estimate, possibly overriding explicitly-set memory settings,
+   * and save useful info (such as the default & estimate) in JobConf for
+   * later consumption.
+   *
+   * Called by Cascading at the start of each job step.
+   */
+  final override def apply(
+    flow: Flow[JobConf],
+    preds: JList[FlowStep[JobConf]],
+    step: FlowStep[JobConf]): Unit = {
+
+    if (skipMemoryEstimation(step)) {
+      LOG.info(s"Skipping memory estimation as ${Config.MemoryEstimators} is not set ")
+    } else {
+      estimate(flow, preds.asScala, step)
+    }
+  }
+
+  private def skipMemoryEstimation(step: FlowStep[JobConf]): Boolean =
+    step.getConfig.get(Config.MemoryEstimators) == null
+
+  private def estimate(
+    flow: Flow[JobConf],
+    preds: Seq[FlowStep[JobConf]],
+    step: FlowStep[JobConf]): Unit = {
+    val conf = step.getConfig
+
+    Option(conf.get(Config.MemoryEstimators)).foreach { clsNames =>
+
+      val clsLoader = Thread.currentThread.getContextClassLoader
+
+      val estimators = StringUtility.fastSplit(clsNames, ",")
+        .map(clsLoader.loadClass(_).newInstance.asInstanceOf[Estimator[MemoryEstimate]])
```

Comment:
how about wrapping the loadClass in a try catch and in case of failures logging an error and failing?